### PR TITLE
Make ScreenLockManager public to fix visibility issue.  Fix a windowing bug.

### PR DESCRIPTION
**Pull request by brandonpage**

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m [hunk 1]
```diff
@@ -549,18 +549,11 @@ - (void)setupServiceConfiguration
 - (void)logoutCleanup
 {
     self.passcodeDisplayed = NO;
-    self.screenLockDisplayed = NO;
 }
 
 - (void)handleAppForeground:(NSNotification *)notification
 {
-    [SFSDKCoreLogger d:[self class] format:@"App is entering the foreground."];
-    if (self.isScreenLockDisplayed) {
```
Comment (brandonpage) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m:558]: This check was carried over from the old passcode system but not needed anymore.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/ScreenLock/ScreenLockManager.swift [hunk 2]
```diff
@@ -28,24 +28,35 @@
...
     
-    @objc func handleAppForground() {
+    // MARK: Screen Lock Manager
+    
+    /// Locks the screen if necessary
+    @objc public func handleAppForground() {
         if readMobilePolicy() {
             lock()
+        } else {
+            unlockPostProcesing()
```
Comment (brandonpage) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/ScreenLock/ScreenLockManager.swift:50]: Not having this was a really bad windowing bug but for some reason isn't necessary if `APPLICATION_EXTENSION_API_ONLY = NO` so I missed it last night.

**Review by wmathurin — APPROVED** [review]

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m [hunk 3]
```diff
@@ -549,18 +549,11 @@ - (void)setupServiceConfiguration
...
-        // Screen Lock was already displayed prior to app foreground.
-        [SFSDKCoreLogger i:[self class] format:@"%@ Screen Lock already displayed.", NSStringFromSelector(_cmd)];
-    } else {
-        [[SFScreenLockManager shared] handleAppForground];
-    }
+    [[SFScreenLockManager shared] handleAppForground];
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m:556]: @brandonpage Sorry for the late comment, but there's a typo on `handleAppForeground`. Could you fix it?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/ScreenLock/ScreenLockManager.swift [hunk 4]
```diff
@@ -28,24 +28,35 @@
...
     private let kScreenLockIdentifier = "com.salesforce.security.screenlock"
     private var callbackBlock: ScreenLockCallbackBlock? = nil
     
     private override init() {}
     
-    @objc func handleAppForground() {
+    // MARK: Screen Lock Manager
+    
+    /// Locks the screen if necessary
+    @objc public func handleAppForground() {
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/ScreenLock/ScreenLockManager.swift:46]: Typo: `handleAppForeground`.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/ScreenLock/ScreenLockManager.swift (hunk 2 above)
Comment (bhariharan) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/ScreenLock/ScreenLockManager.swift:50]: Typo: `unlockPostProcessing`.

**Pull request merged**